src/relay/op/nn/sparse.cc

```diff
@@ -85,10 +117,11 @@ RELAY_REGISTER_OP("nn.sparse_dense")
 )code" TVM_ADD_FILELINE)
     .set_attrs_type<SparseDenseAttrs>()
     .set_num_inputs(4)
-    .add_argument("data", "nD Tensor", "Input data.")
-    .add_argument("weight_data", "1D Tensor", "Weight data matrix.")
-    .add_argument("weight_indices", "1D Tensor", "Weight indices matrix.")
-    .add_argument("weight_indptr", "1D Tensor", "Weight indptr matrix.")
+    .add_argument("input_tensor1", "nD Tensor",
```

Comment:
I think we can not go back to their original names, as the tensor will be used for multiple purposes as given in the description string. I too am not impressed by the name "input_tensor1" :slightly_smiling_face: 
Would appreciate if you can suggest some names.

How about merger of both use case as below? 
"input_tensor1"  => "data_dense_or_sparse"
"input_tensor2"  => "weight_data_or_data_indices"
"input_tensor3"  => "weight_indices_or_data_indptr"
"input_tensor3"  => "weight_indptr_or_dense_weight"

Reply:
I'd prefer `dense_data`, `sparse_data`, `sparse_indices`, `sparse_indptr` as it is less verbose.

Reply:
@tkonolige : Sorry for late response. I think somehow i missed your last reply.
I agree with the naming you suggested, but the issue here is one tensor represents 2 different use case altogether.
For example: 
"input_tensor2" => "weight_data_or_data_indices"
So it can be sparse_data in one case and sparse_indices in another case.
So i am wondering what is the best way to merge both representation ?

Reply:
I would keep the inputs in the original order (no matter what the value of the sparse_data flag is) . That is data, then sparse_data, sparse_indices, sparse_indptr. The order always remains the same, and the flag just changes which computation happens.

Also please change the name of the sparse_data flag. It will conflict with the new names of the parameters.

Reply:
Okay thanks for details, I got your point clearly now!